libs/agno/agno/compression/manager.py

```diff
@@ -0,0 +1,108 @@
+from textwrap import dedent
+from typing import List, Optional
+
+from agno.models.base import Model
+from agno.models.message import Message
+from agno.models.utils import get_model
+from agno.utils.log import log_error, log_warning
+
+DEFAULT_COMPRESSION_PROMPT = dedent("""\
+    You are compressing tool call results to save context space while preserving critical information.
+    
+    Your goal: Extract only the essential information from the tool output.
+    
+    ALWAYS PRESERVE:
+    • Specific facts: numbers, statistics, amounts, prices, quantities, metrics
+    • Temporal data: dates, times, timestamps (use short format: "Oct 21 2025")
+    • Entities: people, companies, products, locations, organizations
+    • Identifiers: URLs, IDs, codes, technical identifiers, versions
+    • Key quotes, citations, sources (if relevant to agent's task)
+    
+    COMPRESS TO ESSENTIALS:
+    • Descriptions: keep only key attributes
+    • Explanations: distill to core insight
+    • Lists: focus on most relevant items based on agent context
+    • Background: minimal context only if critical
+    
+    REMOVE ENTIRELY:
+    • Introductions, conclusions, transitions
+    • Hedging language ("might", "possibly", "appears to")
+    • Meta-commentary ("According to", "The results show")
+    • Formatting artifacts (markdown, HTML, JSON structure)
+    • Redundant or repetitive information
+    • Generic background not relevant to agent's task
+    • Promotional language, filler words
+    
+    EXAMPLE:
+    Input: "According to recent market analysis and industry reports, OpenAI has made several significant announcements in the technology sector. The company revealed ChatGPT Atlas on October 21, 2025, which represents a new AI-powered browser application that has been specifically designed for macOS users. This browser is strategically positioned to compete with traditional search engines in the market. Additionally, on October 6, 2025, OpenAI launched Apps in ChatGPT, which includes a comprehensive software development kit (SDK) for developers. The company has also announced several initial strategic partners who will be integrating with this new feature, including well-known companies such as Spotify, the popular music streaming service, Zillow, which is a real estate marketplace platform, and Canva, a graphic design platform."
+    
+    Output: "OpenAI - Oct 21 2025: ChatGPT Atlas (AI browser, macOS, search competitor); Oct 6 2025: Apps in ChatGPT + SDK; Partners: Spotify, Zillow, Canva"
+    
+    Be concise while retaining all critical facts.
+    """)
+
+
+@dataclass
+class CompressionManager:
+    model: Optional[Model] = None
+    compress_tool_results: bool = True
+    compress_tool_results_limit: int = 3
+    compress_tool_call_instructions: Optional[str] = None
+
+    def _is_tool_result_message(self, msg: Message) -> bool:
+        return msg.role == "tool"
+
+    def should_compress(self, messages: List[Message]) -> bool:
```

Comment:
We can add a log back here (could even be INFO) that just says "compression threshold hit -> Compressing tool results" or something like that